tests/version/version_test.php

```diff
@@ -332,4 +332,204 @@ public function test_get_latest_on_current_branch($current_version, $versions, $
 
 		$this->assertSame($expected, $version_helper->get_latest_on_current_branch());
 	}
+
+	public function get_update_on_branch_data()
+	{
+		return array(
+			array(
+				'1.0.0',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.1',
+					),
+				),
+				array(
+					'current'		=> '1.0.1',
+				),
+			),
+			array(
+				'1.0.1',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.1',
+					),
+				),
+				array(),
+			),
+			array(
+				'1.0.1-a1',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1-a2',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.0',
+					),
+				),
+				array(
+					'current'		=> '1.0.1-a2',
+				),
+			),
+			array(
+				'1.1.0',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.1',
+					),
+				),
+				array(
+					'current'		=> '1.1.1',
+				),
+			),
+			array(
+				'1.1.1',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.1',
+					),
+				),
+				array(),
+			),
+			array(
+				'1.1.0-a1',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.0-a2',
+					),
+				),
+				array(
+					'current'		=> '1.1.0-a2',
+				),
+			),
+			array(
+				'1.1.0',
+				array(),
+				null,
+			),
+			// Latest safe release is 1.0.1
+			array(
+				'1.0.0',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+						'security'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.1',
+					),
+				),
+				array(
+					'current'		=> '1.0.1',
+					'security'		=> '1.0.1',
+				),
+			),
+			// Latest safe release is 1.0.0
+			array(
+				'1.0.0',
+				array(
+					'1.0'	=> array(
+						'current'		=> '1.0.1',
+						'security'		=> '1.0.1',
+					),
+					'1.1'	=> array(
+						'current'		=> '1.1.1',
+					),
+				),
+				array(
+					'current'		=> '1.0.1',
+					'security'		=> '1.0.1',
+				),
+			),
```

Comment:
Fixed below. Github caching is still displaying the old status with it saying `security --> 1.0.1` but it's supplying 1.0.0 now.